addons/sale/models/sale_order.py

```diff
@@ -952,10 +966,10 @@ def _send_order_confirmation_mail(self):
         if self.env.su:
             # sending mail in sudo was meant for it being sent from superuser
             self = self.with_user(SUPERUSER_ID)
-        template_id = self._find_mail_template(force_confirmation_template=True)
-        if template_id:
+        mail_template = self._find_mail_template(force_confirmation_template=True)
+        if mail_template:
             for order in self:
-                order.with_context(force_send=True).message_post_with_template(template_id, composition_mode='comment', email_layout_xmlid="mail.mail_notification_paynow")
+                order.with_context(force_send=True).message_post_with_template(mail_template.id, composition_mode='comment', email_layout_xmlid="mail.mail_notification_paynow")
```

Comment:
You can batch the `message_post_with_template` call as well (for performance purposes)

Reply:
I don't understand what I should do here. Can you give me an example?

Reply:
`self.with_context(force_send=True).message_post_with_template(...)`, the for loop (and multi recordset consideration) can be delegated to `message_post_with_template` bc it supports calls with multiple records in self.

Reply:
I tried your solution but I've got red run bots. I rollbacked to the unbatch solution for now.